cupy/linalg/solve.py

```diff
@@ -54,6 +55,9 @@ def solve(a, b):
     if a.ndim == 2:
         return cupy.cusolver.gesv(a, b)
 
+    if a.shape[-1] <= 256:
```

Comment:
Is this cuBLAS's requirement? Could you add a comment here?

Reply:
Also, if you wanna hijack late (as is done here), you don't have to call `util._assert_cupy_array()` etc in `batched_gesv` and do the array checks twice. Another option is to hijack sooner and leave `batched_gesv` as is.

Reply:
One more question: Is it possible that all of the tests will now go through `batched_gesv` (maybe because the test size is small) and make the main path below untested?

Reply:
This is not a cuBLAS's requirement, but there is a low performance issue in batched getrf routines if matrices are not small..

Reply:
> One more question: Is it possible that all of the tests will now go through batched_gesv (maybe because the test size is small) and make the main path below untested?

Good point! Maybe, we should add tests with large matrices, like a shape of (2, 512, 512) ?

Reply:
I added a routine that allows you to change the threshold to determine whether or not to use batched_gesv.